### ISSUE TITLE
Missing composer dependencies after `docker-compose up`

### ISSUE THREAD

**Issue by naveed125** [naveed125/docker-lumen-app#1]
After running docker-compose up, all the dependencies vanish. Resulting in:

```
% curl -i localhost/api/hello/Naveed
HTTP/1.1 500 Internal Server Error
Server: nginx/1.20.2
Date: Tue, 28 Dec 2021 22:00:42 GMT
Content-Type: text/html; charset=UTF-8
Transfer-Encoding: chunked
Connection: keep-alive
X-Powered-By: PHP/8.0.13
```

**Comment by naveed125** [comment]
Run the following command to restore them, it's caused by mounting the volume in docker-compose, you probably just need to do it once:
```
% docker exec -it -u root docker-lumen-app_web_1 composer install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Package operations: 107 installs, 0 updates, 0 removals
  - Downloading doctrine/inflector (2.0.4)
  - Downloading symfony/polyfill-php72 (v1.23.0)
  - Downloading symfony/polyfill-intl-normalizer (v1.23.0)
  - Downloading symfony/polyfill-intl-idn (v1.23.0)
  - Downloading doctrine/lexer (1.2.1)
  - Downloading egulias/email-validator (2.1.25)
  - Downloading symfony/deprecation-contracts (v3.0.0)
  - Downloading psr/container (1.1.2)
  - Downloading fakerphp/faker (v1.17.0)
  - Downloading symfony/polyfill-php80 (v1.23.1)
  - Downloading symfony/polyfill-mbstring (v1.23.1)
  - Downloading symfony/http-foundation (v5.4.1)
  - Downloading symfony/finder (v5.4.0)
  - Downloading voku/portable-ascii (1.5.6)
  - Downloading symfony/translation-contracts (v3.0.0)
  - Downloading symfony/translation (v6.0.1)
  - Downloading nesbot/carbon (2.55.2)
  - Downloading illuminate/macroable (v8.77.1)
  - Downloading psr/simple-cache (1.0.1)
  - Downloading illuminate/contracts (v8.77.1)
  - Downloading illuminate/collections (v8.77.1)
  - Downloading illuminate/support (v8.77.1)
  - Downloading illuminate/filesystem (v8.77.1)
  - Downloading illuminate/session (v8.77.1)
  - Downloading symfony/polyfill-ctype (v1.23.0)
  - Downloading phpoption/phpoption (1.8.1)
  - Downloading graham-campbell/result-type (v1.0.4)
  - Downloading vlucas/phpdotenv (v5.4.1)
  - Downloading symfony/var-dumper (v5.4.1)
  - Downloading symfony/mime (v5.4.0)
  - Downloading symfony/polyfill-php73 (v1.23.0)
  - Downloading psr/event-dispatcher (1.0.0)
  - Downloading symfony/event-dispatcher-contracts (v3.0.0)
  - Downloading symfony/event-dispatcher (v6.0.1)
  - Downloading psr/log (2.0.0)
  - Downloading symfony/error-handler (v5.4.1)
  - Downloading symfony/http-kernel (v5.4.1)
  - Downloading symfony/polyfill-intl-grapheme (v1.23.1)
  - Downloading symfony/string (v6.0.1)
  - Downloading symfony/service-contracts (v2.4.1)
  - Downloading symfony/console (v5.4.1)
  - Downloading nikic/fast-route (v1.3.0)
  - Downloading illuminate/container (v8.77.1)
  - Downloading illuminate/pipeline (v8.77.1)
  - Downloading illuminate/bus (v8.77.1)
  - Downloading illuminate/events (v8.77.1)
  - Downloading illuminate/view (v8.77.1)
  - Downloading illuminate/translation (v8.77.1)
  - Downloading illuminate/validation (v8.77.1)
  - Downloading illuminate/testing (v8.77.1)
  - Downloading symfony/process (v5.4.0)
  - Downloading symfony/polyfill-php81 (v1.23.0)
  - Downloading ramsey/collection (1.2.2)
  - Downloading brick/math (0.9.3)
  - Downloading ramsey/uuid (4.2.3)
  - Downloading opis/closure (3.6.2)
  - Downloading laravel/serializable-closure (v1.0.5)
  - Downloading illuminate/database (v8.77.1)
  - Downloading illuminate/console (v8.77.1)
  - Downloading illuminate/queue (v8.77.1)
  - Downloading illuminate/pagination (v8.77.1)
  - Downloading monolog/monolog (2.3.5)
  - Downloading illuminate/log (v8.77.1)
  - Downloading illuminate/http (v8.77.1)
  - Downloading illuminate/hashing (v8.77.1)
  - Downloading illuminate/encryption (v8.77.1)
  - Downloading illuminate/config (v8.77.1)
  - Downloading illuminate/cache (v8.77.1)
  - Downloading illuminate/broadcasting (v8.77.1)
  - Downloading illuminate/auth (v8.77.1)
  - Downloading webmozart/assert (1.10.0)
  - Downloading dragonmantank/cron-expression (v3.1.0)
  - Downloading laravel/lumen-framework (v8.3.4)
  - Downloading hamcrest/hamcrest-php (v2.0.1)
  - Downloading mockery/mockery (1.4.4)
  - Downloading phpdocumentor/reflection-common (2.2.0)
  - Downloading phpdocumentor/type-resolver (1.5.1)
  - Downloading phpdocumentor/reflection-docblock (5.3.0)
  - Downloading sebastian/version (3.0.2)
  - Downloading sebastian/type (2.3.4)
  - Downloading sebastian/resource-operations (3.0.3)
  - Downloading sebastian/recursion-context (4.0.4)
  - Downloading sebastian/object-reflector (2.0.4)
  - Downloading sebastian/object-enumerator (4.0.4)
  - Downloading sebastian/global-state (5.0.3)
  - Downloading sebastian/exporter (4.0.4)
  - Downloading sebastian/environment (5.1.3)
  - Downloading sebastian/diff (4.0.4)
  - Downloading sebastian/comparator (4.0.6)
  - Downloading sebastian/code-unit (1.0.8)
  - Downloading sebastian/cli-parser (1.0.1)
  - Downloading phpunit/php-timer (5.0.3)
  - Downloading phpunit/php-text-template (2.0.4)
  - Downloading phpunit/php-invoker (3.1.1)
  - Downloading phpunit/php-file-iterator (3.0.6)
  - Downloading theseer/tokenizer (1.2.1)
  - Downloading nikic/php-parser (v4.13.2)
  - Downloading sebastian/lines-of-code (1.0.3)
  - Downloading sebastian/complexity (2.0.2)
  - Downloading sebastian/code-unit-reverse-lookup (2.0.3)
  - Downloading phpunit/php-code-coverage (9.2.10)
  - Downloading doctrine/instantiator (1.4.0)
  - Downloading phpspec/prophecy (v1.15.0)
  - Downloading phar-io/version (3.1.0)
  - Downloading phar-io/manifest (2.0.3)
  - Downloading myclabs/deep-copy (1.10.2)
  - Downloading phpunit/phpunit (9.5.11)
  - Installing doctrine/inflector (2.0.4): Extracting archive
  - Installing symfony/polyfill-php72 (v1.23.0): Extracting archive
  - Installing symfony/polyfill-intl-normalizer (v1.23.0): Extracting archive
  - Installing symfony/polyfill-intl-idn (v1.23.0): Extracting archive
  - Installing doctrine/lexer (1.2.1): Extracting archive
  - Installing egulias/email-validator (2.1.25): Extracting archive
  - Installing symfony/deprecation-contracts (v3.0.0): Extracting archive
  - Installing psr/container (1.1.2): Extracting archive
  - Installing fakerphp/faker (v1.17.0): Extracting archive
  - Installing symfony/polyfill-php80 (v1.23.1): Extracting archive
  - Installing symfony/polyfill-mbstring (v1.23.1): Extracting archive
  - Installing symfony/http-foundation (v5.4.1): Extracting archive
  - Installing symfony/finder (v5.4.0): Extracting archive
  - Installing voku/portable-ascii (1.5.6): Extracting archive
  - Installing symfony/translation-contracts (v3.0.0): Extracting archive
  - Installing symfony/translation (v6.0.1): Extracting archive
  - Installing nesbot/carbon (2.55.2): Extracting archive
  - Installing illuminate/macroable (v8.77.1): Extracting archive
  - Installing psr/simple-cache (1.0.1): Extracting archive
  - Installing illuminate/contracts (v8.77.1): Extracting archive
  - Installing illuminate/collections (v8.77.1): Extracting archive
  - Installing illuminate/support (v8.77.1): Extracting archive
  - Installing illuminate/filesystem (v8.77.1): Extracting archive
  - Installing illuminate/session (v8.77.1): Extracting archive
  - Installing symfony/polyfill-ctype (v1.23.0): Extracting archive
  - Installing phpoption/phpoption (1.8.1): Extracting archive
  - Installing graham-campbell/result-type (v1.0.4): Extracting archive
  - Installing vlucas/phpdotenv (v5.4.1): Extracting archive
  - Installing symfony/var-dumper (v5.4.1): Extracting archive
  - Installing symfony/mime (v5.4.0): Extracting archive
  - Installing symfony/polyfill-php73 (v1.23.0): Extracting archive
  - Installing psr/event-dispatcher (1.0.0): Extracting archive
  - Installing symfony/event-dispatcher-contracts (v3.0.0): Extracting archive
  - Installing symfony/event-dispatcher (v6.0.1): Extracting archive
  - Installing psr/log (2.0.0): Extracting archive
  - Installing symfony/error-handler (v5.4.1): Extracting archive
  - Installing symfony/http-kernel (v5.4.1): Extracting archive
  - Installing symfony/polyfill-intl-grapheme (v1.23.1): Extracting archive
  - Installing symfony/string (v6.0.1): Extracting archive
  - Installing symfony/service-contracts (v2.4.1): Extracting archive
  - Installing symfony/console (v5.4.1): Extracting archive
  - Installing nikic/fast-route (v1.3.0): Extracting archive
  - Installing illuminate/container (v8.77.1): Extracting archive
  - Installing illuminate/pipeline (v8.77.1): Extracting archive
  - Installing illuminate/bus (v8.77.1): Extracting archive
  - Installing illuminate/events (v8.77.1): Extracting archive
  - Installing illuminate/view (v8.77.1): Extracting archive
  - Installing illuminate/translation (v8.77.1): Extracting archive
  - Installing illuminate/validation (v8.77.1): Extracting archive
  - Installing illuminate/testing (v8.77.1): Extracting archive
  - Installing symfony/process (v5.4.0): Extracting archive
  - Installing symfony/polyfill-php81 (v1.23.0): Extracting archive
  - Installing ramsey/collection (1.2.2): Extracting archive
  - Installing brick/math (0.9.3): Extracting archive
  - Installing ramsey/uuid (4.2.3): Extracting archive
  - Installing opis/closure (3.6.2): Extracting archive
  - Installing laravel/serializable-closure (v1.0.5): Extracting archive
  - Installing illuminate/database (v8.77.1): Extracting archive
  - Installing illuminate/console (v8.77.1): Extracting archive
  - Installing illuminate/queue (v8.77.1): Extracting archive
  - Installing illuminate/pagination (v8.77.1): Extracting archive
  - Installing monolog/monolog (2.3.5): Extracting archive
  - Installing illuminate/log (v8.77.1): Extracting archive
  - Installing illuminate/http (v8.77.1): Extracting archive
  - Installing illuminate/hashing (v8.77.1): Extracting archive
  - Installing illuminate/encryption (v8.77.1): Extracting archive
  - Installing illuminate/config (v8.77.1): Extracting archive
  - Installing illuminate/cache (v8.77.1): Extracting archive
  - Installing illuminate/broadcasting (v8.77.1): Extracting archive
  - Installing illuminate/auth (v8.77.1): Extracting archive
  - Installing webmozart/assert (1.10.0): Extracting archive
  - Installing dragonmantank/cron-expression (v3.1.0): Extracting archive
  - Installing laravel/lumen-framework (v8.3.4): Extracting archive
  - Installing hamcrest/hamcrest-php (v2.0.1): Extracting archive
  - Installing mockery/mockery (1.4.4): Extracting archive
  - Installing phpdocumentor/reflection-common (2.2.0): Extracting archive
  - Installing phpdocumentor/type-resolver (1.5.1): Extracting archive
  - Installing phpdocumentor/reflection-docblock (5.3.0): Extracting archive
  - Installing sebastian/version (3.0.2): Extracting archive
  - Installing sebastian/type (2.3.4): Extracting archive
  - Installing sebastian/resource-operations (3.0.3): Extracting archive
  - Installing sebastian/recursion-context (4.0.4): Extracting archive
  - Installing sebastian/object-reflector (2.0.4): Extracting archive
  - Installing sebastian/object-enumerator (4.0.4): Extracting archive
  - Installing sebastian/global-state (5.0.3): Extracting archive
  - Installing sebastian/exporter (4.0.4): Extracting archive
  - Installing sebastian/environment (5.1.3): Extracting archive
  - Installing sebastian/diff (4.0.4): Extracting archive
  - Installing sebastian/comparator (4.0.6): Extracting archive
  - Installing sebastian/code-unit (1.0.8): Extracting archive
  - Installing sebastian/cli-parser (1.0.1): Extracting archive
  - Installing phpunit/php-timer (5.0.3): Extracting archive
  - Installing phpunit/php-text-template (2.0.4): Extracting archive
  - Installing phpunit/php-invoker (3.1.1): Extracting archive
  - Installing phpunit/php-file-iterator (3.0.6): Extracting archive
  - Installing theseer/tokenizer (1.2.1): Extracting archive
  - Installing nikic/php-parser (v4.13.2): Extracting archive
  - Installing sebastian/lines-of-code (1.0.3): Extracting archive
  - Installing sebastian/complexity (2.0.2): Extracting archive
  - Installing sebastian/code-unit-reverse-lookup (2.0.3): Extracting archive
  - Installing phpunit/php-code-coverage (9.2.10): Extracting archive
  - Installing doctrine/instantiator (1.4.0): Extracting archive
  - Installing phpspec/prophecy (v1.15.0): Extracting archive
  - Installing phar-io/version (3.1.0): Extracting archive
  - Installing phar-io/manifest (2.0.3): Extracting archive
  - Installing myclabs/deep-copy (1.10.2): Extracting archive
  - Installing phpunit/phpunit (9.5.11): Extracting archive
Generating optimized autoload files
62 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
```

**Comment by naveed125** [comment]
After running the above command, everything should be back to normal
```
% curl -i localhost/api/hello/Naveed
HTTP/1.1 200 OK
Server: nginx/1.20.2
Content-Type: application/json
Transfer-Encoding: chunked
Connection: keep-alive
Vary: Accept-Encoding
X-Powered-By: PHP/8.0.13
Cache-Control: no-cache, private
Date: Tue, 28 Dec 2021 22:07:07 GMT

{"hello":"Naveed"}%
```

**Comment by naveed125** [comment]
Resolved with PR #2
